src/main.ts

```diff
@@ -37,17 +55,41 @@ function addCustomPindropFunctionality(map: L.Map): void {
 }
 
 weatherData$.subscribe((data) => {
-  populateCurrentWeather(data.current, data.cityName);
+  populateCurrentWeather(data.current, data.city);
   populateHourlyWeather(data.hourly, data.current);
   showWeatherContainer();
 });
 
 function setWeather(city: CityInterface) {
   updateCitySubject$.next(city);
+  sessionStorage.setItem("currentLat", city.lat.toString());
+  sessionStorage.setItem("currentLng", city.lng.toString());
 }
 
 function addPinsFunctionality(map: L.Map): void {
   cities.forEach((city) => {
```

Comment:
`ForEach` loops are normally not advised. `For/Of` loops are normally preferred. 
That is just something to take into consideration.

Reply:
Thank you

Reply:
I made the changes that you requested. Please have a look at it again.